docker-versions-maven-plugin/src/main/java/com/github/cafapi/docker_versions/plugins/ImageMoniker.java

```diff
@@ -45,14 +47,22 @@ public ImageMoniker(final String repository, final String tag, final String dige
 
         final HostnameReposName hostRepoName = NameParser.resolveRepositoryName(repository);
 
-        this.registry = hostRepoName.hostname;
+        this.registry = getRegistry(hostRepoName);
         this.repositorySansRegistry = hostRepoName.reposName;
```

Comment:
Need to deal with `library/` being optional.  The new goal only works if it is present.